From 343ebc2b82fdfeba845ee27daa6f63127b7822f4 Mon Sep 17 00:00:00 2001
From: Jo-Philipp Wich <jo@mein.io>
Date: Tue, 21 Apr 2026 09:54:33 +0200
Subject: [PATCH] include: register resource type prototypes with VM

In order to ensure reliable freeing of resource type prototypes in complex
prototype chains with per-resource prototypes, register the created type
prototype object with the VM value pool.

Signed-off-by: Jo-Philipp Wich <jo@mein.io>
Upstream-Status: Backport [https://github.com/ucode-lang/ucode/commit/343ebc2b82fdfeba845ee27daa6f63127b7822f4]

---
 include/ucode/lib.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/include/ucode/lib.h
+++ b/include/ucode/lib.h
@@ -86,7 +86,7 @@ uc_resource_new(uc_resource_type_t *type
 static inline uc_resource_type_t *
 _uc_type_declare(uc_vm_t *vm, const char *name, const uc_function_list_t *list, size_t len, void (*freefn)(void *))
 {
-	uc_value_t *proto = ucv_object_new(NULL);
+	uc_value_t *proto = ucv_object_new(vm);
 
 	while (len-- > 0)
 		ucv_object_add(proto, list[len].name,
